openapi.yaml

```
@@ -11717,6 +11717,135 @@ paths:
+      required: true
+      schema:
+        type: string
+    - name: bucket
+      in: path
+      description: The bucket name.
+      required: true
+      schema:
+        type: string
+    get:
+      operationId: getObjectStorageSSL
+      x-linode-cli-action: ssl-view
+      servers:
+      - url: https://api.linode.com/v4
+      summary: View Object Storage SSL Cert
+      description: |
+        Returns a boolean value indicating if this bucket has a corresponding SSL certificate that was
+        uploaded by an Account user.
+      tags:
+      - Object Storage
+      security:
+      - personalAccessToken: []
+      - oauth:
+        - object_storage:read_only
+      responses:
+        '200':
+          description: >
+            Returns a boolean value indicating if this bucket has a corresponding SSL certificate that was
+            uploaded by an Account user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ObjectStorageSSLResponse'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      x-code-samples:
+      - lang: Shell
+        source: >
+          curl -H "Authorization: Bearer $TOKEN" \
+              https://api.linode.com/v4/object-storage/buckets/us-east-1/example-bucket/ssl
+      - lang: CLI
+        source: >
+          linode-cli object-storage ssl-view \
+            us-east-1 example-bucket
+    post:
+      operationId: createObjectStorageSSL
+      x-linode-cli-action: ssl-upload
+      servers:
+      - url: https://api.linode.com/v4
+      summary: Upload Object Storage SSL Cert
+      description: |
+        Upload an SSL certificate and private key to be served when you visit your Object Storage bucket via HTTPS.
+        Your SSL certificate and private key are stored encrypted at rest.
+      tags:
+      - Object Storage
+      security:
+      - personalAccessToken: []
+      - oauth:
+        - object_storage:read_write
+      requestBody:
+        description: Upload this SSL certificate with its corresponding secret key.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ObjectStorageSSL'
+      responses:
+        '200':
+          description: Returns whether this bucket has a corresponding SSL certificate that was uploaded by a user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ObjectStorageSSLResponse'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      x-code-samples:
+      - lang: Shell
+        source: >
+          curl -H "Content-Type: application/json" \
+              -H "Authorization: Bearer $TOKEN" \
+              -X POST -d ‘{
+                  "certificate": """-----BEGIN CERTIFICATE-----MIIDbDCCAlQCCQCl-----END CERTIFICATE-----""",
+                  "private_key": "RwtqMlXveYH531g770BBNluAk9s"
```

Both of these are PEM files so in the json, the newlines will need to be stripped and replaced with "\n"
```suggestion
                  "certificate": "-----BEGIN CERTIFICATE-----\nMIIDbDCCAlQCCQCl\n-----END CERTIFICATE-----",
                  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBADIAQCQI\n-----END PRIVATE KEY-----"
```